### ISSUE TITLE
convert issue from 6.1.71 to 6.4.54 with volume mapping for backup location

### ISSUE THREAD

**Issue by lowrob** [jacobalberty/unifi-docker#476]
### Host operating system
Mint 19.1

### What tag are you using
latest via compose file

### What complete docker command or docker-compose.yml do you use to launch the container (omitting sensitive values)?

Using the compose file.

### What do you expect to happen?
Was going from version 6.1.71 to latest (6.4.54), was expecting a conversion during startup 

### What actually happens?
Appears as part of the startup / conversion first thing that happens is a backup.  Was seeing the following in the logs:
[2021-11-04T02:21:12,085] <launcher> ERROR system - zipDir error
java.io.FileNotFoundException: /usr/lib/unifi/data/backup/6.1.71.unf (Permission denied)
        at java.io.FileOutputStream.open0(Native Method) ~[?:1.8.0_292]
        at java.io.FileOutputStream.open(FileOutputStream.java:270) ~[?:1.8.0_292]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:213) ~[?:1.8.0_292]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:101) ~[?:1.8.0_292]
        at com.ubnt.ace.ooOO.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.FA.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.float.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.float.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.float.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.OO0O.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.nullObject.super(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.oOOO.Ó00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.oOOO.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.C.o00000(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.oOOO.new(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.oOOO.Ó00000(Unknown Source) [ace.jar:?]
        at com.ubnt.ace.Launcher.main(Unknown Source) [ace.jar:?]
[2021-11-04T02:21:12,090] <launcher> ERROR db     - Error while running DB migration
java.lang.IllegalArgumentException: /usr/lib/unifi/data/backup/6.1.71.unf does not exist
        at org.apache.commons.io.FileUtils.sizeOf(FileUtils.java:2541) ~[commons-io-2.6.jar:2.6]
        at com.ubnt.service.system.FA.o00000(Unknown Source) ~[ace.jar:?]

From that I changed the volume section of the docker-compose.yaml file to have the missing location as follows (didn't remove original, don't know if it is still relevant) and the container converted and started fine.:

      # Mount local folder for backups and autobackups
      - ./backup:/unifi/data/backup
      - ./backup2:/usr/lib/unifi/data/backup
Prior to this the controller container just kept restarting over and over.

**Comment by regnete** [comment]
We are facing the same effect. @lowrob Your workaround didn't fix the issue for us. The container ist still restarting every 20 seconds.

We are using the latest version via compose file. Looks like the container is trying a downgrade from  6.4.54 to 6.5.53. Did anybody change the latest tag from 6.5.54 to 6.5.53?

```
[2021-12-01T11:34:24,942] <launcher> INFO  tomcat - Adding basic REST API support during the startup
[2021-12-01T11:34:25,252] <launcher> INFO  system - Tomcat startup took 546ms
[2021-12-01T11:34:25,253] <launcher> INFO  db     - Version 6.4.54 has changed to 6.5.53, attempting to start migration
[2021-12-01T11:34:25,422] <launcher> INFO  system - ======================================================================
[2021-12-01T11:34:25,422] <launcher> INFO  system - UniFi 6.5.53 (build atag_6.5.53_16673 - release/release) is started
[2021-12-01T11:34:25,422] <launcher> INFO  system - ======================================================================
[2021-12-01T11:34:25,423] <launcher> INFO  system - BASE dir:/usr/lib/unifi
[2021-12-01T11:34:25,453] <launcher> INFO  system - Current System IP: 192.168.240.3
[2021-12-01T11:34:25,453] <launcher> INFO  system - Hostname: controller
[2021-12-01T11:34:25,453] <launcher> INFO  system - ubic.env: prod
[2021-12-01T11:34:25,553] <launcher> INFO  db     - waiting for db connection...
[2021-12-01T11:34:26,054] <launcher> INFO  db     - Connecting to mongodb://mongo/unifi
[2021-12-01T11:34:27,472] <launcher> INFO  db     - Connecting to mongodb://mongo/unifi_stat
[2021-12-01T11:34:27,828] <launcher> INFO  productinfo - Using controller channel=RELEASE, firmware channel=RELEASE. Available controller channels=[RELEASE], available firmware channels=[RELEASE]. SSO is disabled.
[2021-12-01T11:34:27,965] <launcher> INFO  webrtc - WebRTC library version: EvoStream Media Server (www.evostream.com) build v2.9.1 - Gladiator - (built for Debian-8.2.0-x86_64 on 2020-02-05T23:33:19.000) OpenSSL version: 1.1.1d usrsctp version: v0.1.2 compiled on machine: Linux debian-8-2-0-64 3.16.0-6-amd64 #1 SMP Debian 3.16.57-2 (2018-07-14) x86_64 GNU/Linux
[2021-12-01T11:34:29,280] <launcher> ERROR system - zipDir error
java.io.FileNotFoundException: /usr/lib/unifi/data/backup/6.4.54.unf (Permission denied)
        at java.io.FileOutputStream.open0(Native Method) ~[?:1.8.0_292]
        at java.io.FileOutputStream.open(FileOutputStream.java:270) ~[?:1.8.0_292]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:213) ~[?:1.8.0_292]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:101) ~[?:1.8.0_292]
        at com.ubnt.ace.OoOO.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.FA.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.M.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.OooOo.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ó00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.C.super(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ò00000(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ó00000(Unknown Source) [ace.jar:?]
        at com.ubnt.ace.Launcher.main(Unknown Source) [ace.jar:?]
[2021-12-01T11:34:29,285] <launcher> ERROR db     - Error while running DB migration
java.lang.IllegalArgumentException: /usr/lib/unifi/data/backup/6.4.54.unf does not exist
        at org.apache.commons.io.FileUtils.sizeOf(FileUtils.java:2541) ~[commons-io-2.6.jar:2.6]
        at com.ubnt.service.system.FA.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.L.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.backup.M.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.OooOo.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ó00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.o00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.C.super(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ò00000(Unknown Source) [ace.jar:?]
        at com.ubnt.service.system.db.migrator.startup.A.Ó00000(Unknown Source) [ace.jar:?]
        at com.ubnt.ace.Launcher.main(Unknown Source) [ace.jar:?]
[2021-12-01T11:34:29,322] <pool-2-thread-1> ERROR Asynchronous EventBusService - Exception thrown by subscriber method super(com.ubnt.service.devmgr.device.ws.A.while) on subscriber com.ubnt.service.devmgr.device.ws.super@42403dc6 when dispatching event: DeviceUpgradeStateUpdateEvent{device={"_id":"5baa991cb036890036b849ed","ip":"192.168.10.132","mac":"fc:ec:da:f0:57:ab","model":"U7IW","type":"uap","version":"5.43.52.12774","adopted":true,"site_id":"5b5c0e74e64ada00328bd8c5","cfgversion":"2cb9424176e5c206","config_network":{"type":"dhcp","ip":"192.168.10.132"},"vwire_table":[],"license_state":"registered","inform_url":"http://unifi.sitewaerts.de:8080/inform","inform_ip":"188.64.61.160","board_rev":19,"antenna_table":[{"default":true,"id":4,"name":"Combined","wifi0_gain":1,"wifi1_gain":2}],"radio_table":[{"radio":"ng","name":"wifi0","max_txpower":20,"min_txpower":4,"nss":2,"radio_caps":16404,"radio_caps2":27,"builtin_antenna":true,"builtin_ant_gain":1,"current_antenna_gain":0},{"radio":"na","name":"wifi1","max_txpower":20,"min_txpower":4,"nss":2,"is_11ac":true,"has_dfs":true,"has_fccdfs":true,"radio_caps":50479124,"radio_caps2":27,"builtin_antenna":true,"builtin_ant_gain":2,"current_antenna_gain":0}],"scan_radio_table":[],"ethernet_table":[{"mac":"fc:ec:da:f0:57:ab","num_port":3,"name":"eth0"}],"port_table":[{"port_idx":1,"media":"GE","port_poe":false,"poe_caps":16},{"port_idx":2,"media":"GE","port_poe":false,"poe_caps":0},{"port_idx":3,"media":"GE","port_poe":false,"poe_caps":0,"attr_no_edit":true}],"has_speaker":false,"has_eth1":false,"fw_caps":67042623,"hw_caps":0,"wifi_caps":5160917,"led_override":"default","led_override_color":"#0000ff","led_override_color_brightness":100,"name":"AC IW Arbeit","outdoor_mode_override":"default","serial":"FCECDAF057AB","countrycode_table":[],"bandsteering_mode":"prefer_5g","wlangroup_id_na":"5b5c0e74e64ada00328bd8ce","wlangroup_id_ng":"5b5c0e74e64ada00328bd8ce","required_version":"3.7.12","map_id":"5b5c0e74e64ada00328bd8d6","x":51.07352681331686,"y":8.095364625343564,"unsupported":false,"unsupported_reason":0,"sys_error_caps":0,"has_fan":false,"has_temperature":false,"two_phase_adopt":false,"syslog_key":"129777c27f06d0f39a742fbe0623ad8f02df3479c118ce36b75b4152ea30c7b1","kernel_version":"4.4.153","architecture":"mips","hash_id":"7a7036c9c071db52","gateway_mac":"fc:ec:da:43:e0:5a","manufacturer_id":4,"wlan_overrides":[{"wlan_id":"5b5c0efbe64ada00328bd8dc","radio":"ng","radio_name":"wifi0","enabled":false},{"wlan_id":"5b9a302400be87003672c16b","radio":"ng","radio_name":"wifi0","enabled":false}],"internet":true,"switch_caps":{"feature_caps":32,"max_mirror_sessions":0,"max_aggregate_sessions":0,"max_l3_intf":0,"max_reserved_routes":0,"max_static_routes":0},"country_code":0,"model_incompatible":false,"model_in_lts":false,"model_in_eol":false,"connected_at":1637500534,"provisioned_at":1637373775,"anon_id":"61504cce-5463-4a77-8a70-36c9c071db52","atf_enabled":true,"last_uplink":{"uplink_mac":"74:ac:b9:a3:a7:52","uplink_remote_port":4,"port_idx":3,"type":"wire"},"supports_fingerprint_ml":false,"device_id":"5baa991cb036890036b849ed","uplink":{"uplink_mac":"74:ac:b9:a3:a7:52","uplink_remote_port":4,"port_idx":3,"type":"wire"},"state":0,"start_disconnected_millis":1638358469180,"upgrade_state":0}, upgradeState=0}
org.springframework.beans.factory.BeanCreationNotAllowedException: Error creating bean with name 'staManager': Singleton bean creation not allowed while singletons of this factory are in destruction (Do not request a bean from a BeanFactory in a destroy method implementation!)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:220) ~[spring-beans-5.3.7.jar:5.3.7]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.7.jar:5.3.7]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.7.jar:5.3.7]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.7.jar:5.3.7]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.7.jar:5.3.7]
        at org.springframework.context.annotation.ContextAnnotationAutowireCandidateResolver$1.getTarget(ContextAnnotationAutowireCandidateResolver.java:95) ~[spring-context-5.3.7.jar:5.3.7]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:676) ~[spring-aop-5.3.7.jar:5.3.7]
        at com.ubnt.service.devmgr.v$$EnhancerBySpringCGLIB$$6674e427.ÒÖ0000(<generated>) ~[ace.jar:?]
        at com.ubnt.service.devmgr.live.O0OO.super(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.live.O0OO.super(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.live.C.super(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.device.ws.super.Ò00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.device.ws.super.super(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.device.ws.super.Ò00000(Unknown Source) ~[ace.jar:?]
        at com.ubnt.service.devmgr.device.ws.super.super(Unknown Source) ~[ace.jar:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87) ~[guava-27.1-jre.jar:?]
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144) ~[guava-27.1-jre.jar:?]
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72) [guava-27.1-jre.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]

```

**Comment by jacobalberty** [comment]
@regnete there never was a version 6.5.54, 6.5.53 is the latest version in early access or released. It looks like you have a simple permission error. Make sure your unifi volumes are owned by userid 999 and group id 998 and writeable by that user.

**Comment by regnete** [comment]
Thanks for the quick reply. I already recognized missing privileges and fixed them. That solved one oprt of the issue.

But: All I did was `docker-compose pull && docker-compose down && docker-compose up -d`. After that I ended up with an endless restarting container.

I will check now, which image versions I can extract from the local docker cache. May be that gives a hint why we were facing a downgrade situation.
